ci-operator/step-registry/aws/provision/vpc/shared/dualstack/aws-provision-vpc-shared-dualstack-commands.sh

```diff
@@ -0,0 +1,697 @@
+                                !Join [ "+", [ !Ref PrivateSubnet3 ] ]
+                              ]
+                            ]
+                    ]
+                  ],
+            ""
+          ]
+EOF
+# In the above template, note that:
+# PublicRoute.DestinationIpv6CidrBlock (::/0->InternetGateway) make the instances in public subnets have the capacity
+# of outgoing traffic to internet over IPv6, or else, they can't go to internet even if they have ipv6 address.
+# If want the instances in private subnets also have the capacity of outgoing traffic to internet over IPv6, have to
+# create AWS::EC2::EgressOnlyInternetGateway, then add route rules (::/0->EgressOnlyInternetGateway) in each private
+# subnet's Route-N. EgressOnlyInternetGateway is only for ipv6, which is similar to NAT only for ipv6. We can utilize
+# it to create a disconnected ipv6 network environment.
+# IPv6-based service running on the instance in public subnets can be accessed, but if the instance is running in a
+# private subnet, the service can not be available.
+
+MAX_ZONES_COUNT=$(aws --region "${REGION}" ec2 describe-availability-zones --filter Name=state,Values=available Name=zone-type,Values=availability-zone | jq '.AvailabilityZones | length')
+if (( ZONES_COUNT > MAX_ZONES_COUNT )); then
+  ZONES_COUNT=$MAX_ZONES_COUNT
+fi
+
+# The above cloudformation template's max zones account is 3
+if [[ "${ZONES_COUNT}" -gt 3 ]]; then
+  ZONES_COUNT=3
+fi
+
+STACK_NAME="${NAMESPACE}-${UNIQUE_HASH}-vpc"
+echo ${STACK_NAME} >> "${SHARED_DIR}/to_be_removed_cf_stack_list"
+
+vpc_params="${ARTIFACT_DIR}/vpc_params.json"
+aws_add_param_to_json "AvailabilityZoneCount" ${ZONES_COUNT} "$vpc_params"
+
+if [[ -n "${VPC_CIDR}" ]]; then
+     aws_add_param_to_json "VpcCidr" ${VPC_CIDR} "$vpc_params"
+fi
+
+if [[ ${ZONES_LIST} != "" ]]; then
+  zones_list_count=$(echo "$ZONES_LIST" | awk -F',' '{ print NF }')
+  if [[ "${zones_list_count}" != "${ZONES_COUNT}" ]]; then
+    echo "ERROR: ${zones_list_count} zones in the list [${ZONES_LIST}], the zone count in the list should be the same as ZONES_COUNT: ${ZONES_COUNT}, exit now"
+    exit 1
+  fi
+  aws_add_param_to_json "AllowedAvailabilityZoneList" "${ZONES_LIST}" "$vpc_params"
+fi
+
+aws --region "${REGION}" cloudformation create-stack \
+  --stack-name "${STACK_NAME}" \
+  --template-body "$(cat /tmp/01_vpc.yaml)" \
+  --tags "${TAGS}" \
+  --parameters file://${vpc_params} &
+
+wait "$!"
+echo "Created stack"
+
+aws --region "${REGION}" cloudformation wait stack-create-complete --stack-name "${STACK_NAME}" &
+wait "$!"
+echo "Waited for stack"
+
+# save vpc stack output
+aws --region "${REGION}" cloudformation describe-stacks --stack-name "${STACK_NAME}" > "${SHARED_DIR}/vpc_stack_output"
+
+# save vpc id
+# e.g. 
+#   vpc-01739b6510a152d44
+VpcId=$(jq -r '.Stacks[].Outputs[] | select(.OutputKey=="VpcId") | .OutputValue' "${SHARED_DIR}/vpc_stack_output")
+echo "$VpcId" > "${SHARED_DIR}/vpc_id"
```

Comment:
How about using new `vpc_info_json=${SHARED_DIR}/vpc_info.json` to replace the this legacy way completely. (we may need another PR to adopt vpc_info.json to existing jobs)

Reply:
okay, I am going to drop subnets id and az files together.